Assignment6/Q1.py

```diff
@@ -0,0 +1,103 @@
+            are values.
+    """
+    position_map = {}
+    for i in range(len(current_positions)):
+        car = current_positions[i]
+        position_map[car] = i
+    return position_map
+
+
+def make_move(car, start_spot, end_spot, current_positions, position_map, moves):
+    """The method 'performs' a move, i.e. adds move to the move list and updates all position data structures.
+
+        Args:
+            car: an integer, indicates a car on a parking
+            start_spot: an integer, the space where the move 'begins'
+            end_spot: an integer, the space where the move 'ends'
+            current_positions: an array of integers, indicates the current locations (layout) of cars on the parking
+            position_map: a dictionary mapping integers to integers, cars' numbers are keys and cars' position on the
+                          parking are values
+            moves: an array of Move objects, sequence of the moves 'done so far'
+    """
+    moves.append(Move(start_spot, end_spot))
+    position_map[0] = start_spot
+    position_map[car] = end_spot
+    current_positions[start_spot] = 0
+    current_positions[end_spot] = car
+
+
+def set_single_spot(spot, current_positions, position_map, desired_positions, moves):
+    """The method sets a 'desired car' on the spot passed as an argument.
+       It uses the maximum of two moves to set the car.
+
+        Args:
+            spot: an integer, the parking space we want to set the car on
+            current_positions: an array of integers, indicates the current locations (layout) of cars on the parking
+            position_map: a dictionary mapping integers to integers, cars' numbers are keys and cars' position on the
+                          parking are values
+            desired_positions: an array of integers, indicates the desired locations (layout) of cars on the parking
+            moves: an array of Move objects, sequence of the moves 'done so far'
+    """
+    current_car = current_positions[spot]
+    desired_car = desired_positions[spot]
+
+    # There is no need to set a spot if the right car is already there.
+    if current_car != desired_car:
+        # The first move is to empty the spot. There is no need to empty the spot if it is already empty (not occupied).
+        if current_car != 0:
+            empty_spot = position_map[0]
+            make_move(car=current_car, start_spot=spot, end_spot=empty_spot, current_positions=current_positions,
+                      position_map=position_map, moves=moves)
+
+        # The second move sets to right car to now empty parking space.
+        desired_car_spot = position_map[desired_car]
+        make_move(car=desired_car, start_spot=desired_car_spot, end_spot=spot, current_positions=current_positions,
+                  position_map=position_map, moves=moves)
+
+
+def find_moves_sequence(start_positions, desired_positions, print_sequence):
```

Comment:
I think it would be more understandable, if it will be the first function if this file.